### PR TITLE
Add Verticies to building maps

### PR DESCRIPTION
**Pull request by S3JER**
Adds verticies to building maps and exposes the map's verticies to robot controller.
The verticies are generated based on a home-cooked algorithm that is very bad and should be replaced asap.
This is a prerequisite for writing patrolling algorithms.

This still uses the exploration stopping criteria.
TODO: Should we later separate exploration and patrolling completely?

### CODE REVIEW

File: Assets/Scripts/Map/Vertex.cs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+namespace Maes.Map {
+    public class Vertex
+    {
+        private HashSet<Vertex> _neighbors = new HashSet<Vertex>();
+        private float _weight;
+        private int _idleness;
+        private Vector2Int _position;
+        private int _totalIdleness;
+        private int _visits;
+        private float _maxIdleness;
```
Comment (Puvikaran2001) [inline Assets/Scripts/Map/Vertex.cs]: Replace float with int

File: Assets/Scripts/Statistics/PatrollingVisualizer.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+        public GameObject EdgeVisualizer;
+
+        private PatrollingMap _patrollingMap;
+
+        private List<GameObject> _visualizers = new List<GameObject>();
+
+        public void SetMap(SimulationMap<Tile> simulationMap)
+        {
+            // Why the fuck do i have to do this?
```
Comment (Puvikaran2001) [inline Assets/Scripts/Statistics/PatrollingVisualizer.cs:20]: I think we need to either remove the comment or write something else :)

File: Assets/Scripts/WallFollowerExperiment.cs [hunk 3]
```diff
@@ -138,7 +138,7 @@ private void Start()
                     { "wall_follower", seed => new WallFollowerAlgorithm() },
                 };
             constraintIterator++;
-            var buildingMaps = buildingConfigList50.Union(buildingConfigList75.Union(buildingConfigList100));
+            var buildingMaps = ((buildingConfigList100));
```
Comment (Puvikaran2001) [inline Assets/Scripts/WallFollowerExperiment.cs:141]: Remove the parentheses :)

File: Assets/Scripts/Map/PatrollingMap.cs [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+                        firstTri.Type == TileType.Wall
+                        || firstTri.Type == TileType.Brick
+                        || firstTri.Type == TileType.Concrete
+                        || firstTri.Type == TileType.Wood
+                    )
+                    {
+                        continue;
+                    }
+
+                    roomTiles.Add(new Vector2Int(x, y));
```
Comment (Puvikaran2001) [inline Assets/Scripts/Map/PatrollingMap.cs:36]:
```suggestion
                    if (!Tile.IsWall(firstTri.Type))
                    {
                        roomTiles.Add(new Vector2Int(x, y));
                    }
```

**Review by Puvikaran2001 — CHANGES_REQUESTED** [review]

**Review by mads256h — APPROVED** [review]
LGTM from me but i wrote some of this soooo

File: Assets/Scripts/Statistics/PatrollingVisualizer.cs (hunk 2 above)
Comment (mads256h) [reply Assets/Scripts/Statistics/PatrollingVisualizer.cs]: Better?

**Review by Puvikaran2001 — APPROVED** [review]

**Pull request merged**